applications/HDF5Application/python_scripts/core/operations/model_part.py

```diff
@@ -299,104 +325,132 @@ class PrimalBossakOutput(VariableIO):
+        self.__alpha_bossak = parameters["alpha_bossak"].GetDouble()
+        super().__init__(model_part, parameters, file)
+
+    @property
+    def alpha_bossak(self) -> float:
+        return self.__alpha_bossak
 
-    def __call__(self, model_part, hdf5_file):
-        primal_io = KratosHDF5.HDF5NodalSolutionStepBossakIO(
-            self.GetSettings(model_part).Get(), hdf5_file)
-        primal_io.SetAlphaBossak(self.alpha_bossak)
-        primal_io.WriteNodalResults(model_part)
+    def Execute(self) -> None:
+        primal_io = KratosHDF5.HDF5NodalSolutionStepBossakIO(self.parameters, self.file)
+        primal_io.SetAlphaBossak(self.__alpha_bossak)
+        primal_io.WriteNodalResults(self.model_part)
 
+    @staticmethod
+    def GetDefaultParameters() -> KratosMultiphysics.Parameters:
+        parameters = super(PrimalBossakOutput, PrimalBossakOutput).GetDefaultParameters()
+        parameters.AddDouble("alpha_bossak", -0.3)
+        return parameters
 
-class PrimalBossakInput(VariableIO):
+
+class PrimalBossakInput(VariableIOOperation):
     '''Reads nodal solution step data from a file.
 
     This is used by the transient adjoint solvers.
     '''
 
-    def __init__(self, settings):
-        super(PrimalBossakInput, self).__init__(settings)
-
-    def __call__(self, model_part, hdf5_file):
-        primal_io = KratosHDF5.HDF5NodalSolutionStepBossakIO(
-            self.GetSettings(model_part).Get(), hdf5_file)
-        primal_io.ReadNodalResults(
-            model_part)
+    def Execute(self) -> None:
+        KratosHDF5.HDF5NodalSolutionStepBossakIO(
+            self.parameters,
+            self.file).ReadNodalResults(self.model_part)
 
 
-class MoveMesh:
+class MoveMesh(ModelPartIOOperation):
     '''Perform a mesh move operation on a model part.
 
     The primary use case is to set the mesh to the current configuration after
     reading the model part.
     '''
 
-    def __call__(self, model_part, *args):
-        KratosMultiphysics.ImplicitSolvingStrategy(model_part, True).MoveMesh()
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        print("HDF5 MoveMesh operation is deprecated and will be removed", file = sys.stderr)
 
+    def Execute(self) -> None:
+        KratosMultiphysics.ImplicitSolvingStrategy(self.model_part, True).MoveMesh()
 
-def Create(settings):
-    '''Return the operation specified by the setting 'operation_type'.
+
+def GetSubclasses(base_class: type) -> "list[type]":
+    """Recursively find all subclasses of a base class"""
+    subclasses = base_class.__subclasses__()
+    for subclass in base_class.__subclasses__():
+        subclasses += GetSubclasses(subclass)
+    return subclasses
+
+
+class AggregateOperation(KratosMultiphysics.Operation):
+    """ @brief Class for aggregating HDF5 IO operations on the same file."""
+
+    def __init__(self, model: KratosMultiphysics.Model, parameters: typing.Union[KratosMultiphysics.Parameters, ParametersWrapper]):
```

Comment:
May be in future we can get rid of the `ParameterWrapper`. :D

Reply:
I'd very much like that. `ParametersWrapper` is a neat idea but unless `Parameters` gets replaced by it in every python script, it's just a pain to deal with.